### DISCUSSION TITLE
cQube release 3.3 Installation failed with ansible error

### DISCUSSION THREAD

**Discussion by srajasimman** [Sunbird-cQube/community#35] [Q&A] [answered]
Hi @SreenivasNimmagadda,

I have created config.yml (#33) as follows for cQube-Workflow release 3.3 installation.
```yml
---
base_dir: /opt                            
state_code: GJ                            
diksha_columns: false                     
static_datasource: udise                  
management: govt                          
session_timeout: 7D                       
map_name: leafletmap                      
google_api_key: remove_this_key_value     
 
theme: theme1                             
slab1: 35                                 
slab2: 35-60                              
slab3: 60-80                              
slab4: 80           
```
The installation was failed with the following error.
```python
TASK [Run diksha python code] *****************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {
  "changed": true,
  "cmd": "/opt/cqube/emission_app/flaskenv/bin/python /opt/cqube/emission_app/python/deploy_nifi.py diksha_transformer diksha_parameters 4662",
  "delta": "0:00:01.150416",
  "end": "2022-02-08 18:33:37.156519",
  "msg": "non-zero return code",
  "rc": 1,
  "start": "2022-02-08 18:33:36.006103",
  "stderr": "INFO:root:Uploading Template to Nifi......\nDEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost:8096\nDEBUG:urllib3.connectionpool:http://localhost:8096 \"GET /nifi-api/process-groups/root HTTP/1.1\" 200 626\nDEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost:8096\nDEBUG:urllib3.connectionpool:http://localhost:8096 \"POST /nifi-api/process-groups/d96abf65-017e-1000-9168-ae9dbe5f2489/templates/upload HTTP/1.1\" 201 278\nINFO:root:Successfully uploaded the template.\nINFO:root:Instatiating the template.\nDEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost:8096\nDEBUG:urllib3.connectionpool:http://localhost:8096 \"GET /nifi-api/flow/templates HTTP/1.1\" 200 774\nDEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost:8096\nDEBUG:urllib3.connectionpool:http://localhost:8096 \"POST /nifi-api/process-groups/d96abf65-017e-1000-9168-ae9dbe5f2489/template-instance HTTP/1.1\" 201 731\nINFO:root:Successfully Instatiated the template.\nINFO:root:Reading dynamic parameters from file diksha_parameters.json\nINFO:root:Reading static parameters from file diksha_parameters.txt\nTraceback (most recent call last):\n  File \"/opt/cqube/emission_app/python/deploy_nifi.py\", line 302, in <module>\n    parameter_body=update_nifi_params.nifi_params_config(parameter_context_name, f'{prop.NIFI_STATIC_PARAMETER_DIRECTORY_PATH}{params.get(parameter_context_name)}',parameter_body)\n  File \"/opt/cqube/emission_app/python/update_nifi_params.py\", line 11, in nifi_params_config\n    parameter_dict = ast.literal_eval(parameter_data)\n  File \"/usr/lib/python3.6/ast.py\", line 48, in literal_eval\n    node_or_string = parse(node_or_string, mode='eval')\n  File \"/usr/lib/python3.6/ast.py\", line 35, in parse\n    return compile(source, filename, mode, PyCF_ONLY_AST)\n  File \"<unknown>\", line 11\n    \"diksha_tpd_encryption\":'''<Type true if tpd file is password proctected else type false. Ex: \"diksha_tpd_encryption\":'''true''' >''',\n                                                                                                                                ^\nSyntaxError: invalid syntax",
  "stderr_lines": [
    "INFO:root:Uploading Template to Nifi......",
    "DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost:8096",
    "DEBUG:urllib3.connectionpool:http://localhost:8096 \"GET /nifi-api/process-groups/root HTTP/1.1\" 200 626",
    "DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost:8096",
    "DEBUG:urllib3.connectionpool:http://localhost:8096 \"POST /nifi-api/process-groups/d96abf65-017e-1000-9168-ae9dbe5f2489/templates/upload HTTP/1.1\" 201 278",
    "INFO:root:Successfully uploaded the template.",
    "INFO:root:Instatiating the template.",
    "DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost:8096",
    "DEBUG:urllib3.connectionpool:http://localhost:8096 \"GET /nifi-api/flow/templates HTTP/1.1\" 200 774",
    "DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost:8096",
    "DEBUG:urllib3.connectionpool:http://localhost:8096 \"POST /nifi-api/process-groups/d96abf65-017e-1000-9168-ae9dbe5f2489/template-instance HTTP/1.1\" 201 731",
    "INFO:root:Successfully Instatiated the template.",
    "INFO:root:Reading dynamic parameters from file diksha_parameters.json",
    "INFO:root:Reading static parameters from file diksha_parameters.txt",
    "Traceback (most recent call last):",
    "  File \"/opt/cqube/emission_app/python/deploy_nifi.py\", line 302, in <module>",
    "    parameter_body=update_nifi_params.nifi_params_config(parameter_context_name, f'{prop.NIFI_STATIC_PARAMETER_DIRECTORY_PATH}{params.get(parameter_context_name)}',parameter_body)",
    "  File \"/opt/cqube/emission_app/python/update_nifi_params.py\", line 11, in nifi_params_config",
    "    parameter_dict = ast.literal_eval(parameter_data)",
    "  File \"/usr/lib/python3.6/ast.py\", line 48, in literal_eval",
    "    node_or_string = parse(node_or_string, mode='eval')",
    "  File \"/usr/lib/python3.6/ast.py\", line 35, in parse",
    "    return compile(source, filename, mode, PyCF_ONLY_AST)",
    "  File \"<unknown>\", line 11",
    "    \"diksha_tpd_encryption\":'''<Type true if tpd file is password proctected else type false. Ex: \"diksha_tpd_encryption\":'''true''' >''',",
    "                                                                                                                                ^",
    "SyntaxError: invalid syntax"
  ],
  "stdout": "",
  "stdout_lines": []
}

PLAY RECAP ************************************************************************************************************************************************************************
localhost                  : ok=195  changed=150  unreachable=0    failed=1    skipped=6    rescued=0    ignored=0
```

Please check and let me if you need any more information.

cc: @Ramya-kmurthy, @LavanyaKR3007, @gandham-santhosh

Regards,
Rajasimman S

**Comment by jp-tibil** [comment, marked as answer]
Hi Rajasimman, 

This error is because the diksha_parameters file was not filled.
If you set 'true' for diksha datasource in this [datasource_config.yml](https://github.com/Sunbird-cQube/cQube_Workflow/blob/cQube-development/workflow_deploy/education_usecase/datasource_config.yml) file, then you need to fill the required parameters in this [diksha_parameters.txt](https://github.com/Sunbird-cQube/cQube_Workflow/blob/cQube-development/development/datasource/diksha/nifi/diksha_parameters.txt) file. 

To skip diksha datasource, you can set 'false' for diksha_datasource in [datasource_config.yml](https://github.com/Sunbird-cQube/cQube_Workflow/blob/cQube-development/workflow_deploy/education_usecase/datasource_config.yml) file and re-run the installation.

Thanks,
Jeyaprakash Ramaraj

**Reply by srajasimman** [reply]
Hi Jayaprakash,

Thanks for the reply. I will check  and update you if any

Regards,
Rajasimman

**Reply by gandham-santhosh** [reply]
@jp-tibil can you please help to understand the values for the variables mentioned in the config file.  Also, please let us know if we need to change any other content in the file apart from the below mentioned variables

- "diksha_api_progress_exhaust_api_token":''' '''
    - Is this API Token for http://docs.sunbird.org/latest/apis/dataexhaustapi/index.html#tag/Public-Data-Exhaust-API(s) ? If not please let us which API it is referring to 
- "diksha_api_progress_exhaust_batch_id_list":''' ''',
   - Can we give comma separated batch id's here? If not please share a sample value for this variable  
-  "diksha_api_progress_exhaust_dataset":''' ''',
   - Can you please let us know what should be the value for this variable?  
-  "diksha_api_progress_exhaust_encryption_key":''' ''',
   - Can this be empty? If not please let us know what  should be the value for this variable
- "diksha_api_progress_exhaust_x_channel_id":''' ''',
   - This is the channel id for which we want to generate the reports for. Please let us know if our understanding is correct  
- "diksha_api_summary_roll_up_api_token":''' ''',
   - Is this the API token for http://docs.sunbird.org/latest/apis/reports/#tag/Report-Summary API? If not please let us know which API it is referring to
- "diksha_api_summary_roll_up_x_channel_id":''' ''',
   - Which channel Id we should provide as input here?

**Comment by jai-ekstep** [comment]
It would be easier to do the install with Diksha data initially and
then add Diksha data to the existing installation<div
id="DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2"><br />
<table style="border-top: 1px solid #D3D4DE;">
	<tr>
        <td style="width: 55px; padding-top: 13px;"><a
href="https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail"
target="_blank"><img
src="https://ipmcdn.avast.com/images/icons/icon-envelope-tick-round-orange-animated-no-repeat-v1.gif"
alt="" width="46" height="29" style="width: 46px; height: 29px;"
/></a></td>
		<td style="width: 470px; padding-top: 12px; color: #41424e;
font-size: 13px; font-family: Arial, Helvetica, sans-serif;
line-height: 18px;">Virus-free. <a
href="https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail"
target="_blank" style="color: #4453ea;">www.avast.com</a>
		</td>
	</tr>
</table><a href="#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2" width="1"
height="1"></a></div>

**Reply by gandham-santhosh** [reply]
@jai-ekstep do you mean without DIKSHA data first and then try with DIKSHA data later?

**Comment by jai-ekstep** [comment]
Yes
So the configyration for Diksha will not need to be done<div
id="DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2"><br />
<table style="border-top: 1px solid #D3D4DE;">
	<tr>
        <td style="width: 55px; padding-top: 13px;"><a
href="https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail"
target="_blank"><img
src="https://ipmcdn.avast.com/images/icons/icon-envelope-tick-round-orange-animated-no-repeat-v1.gif"
alt="" width="46" height="29" style="width: 46px; height: 29px;"
/></a></td>
		<td style="width: 470px; padding-top: 12px; color: #41424e;
font-size: 13px; font-family: Arial, Helvetica, sans-serif;
line-height: 18px;">Virus-free. <a
href="https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail"
target="_blank" style="color: #4453ea;">www.avast.com</a>
		</td>
	</tr>
</table><a href="#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2" width="1"
height="1"></a></div>
